src/main/java/com/example/purerestapi/repository/UserRepository.java

```diff
@@ -0,0 +1,36 @@
+package com.example.purerestapi.repository;
+
+import com.example.purerestapi.entity.User;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class UserRepository {
+
+  private final static Map<String, User> users = new HashMap<>();
```

Comment:
It is better to use ConcurrentHashMap, since HashMap is not thread-safe

Reply:
agree